src/re.rs

```diff
@@ -1025,19 +1044,16 @@ impl<'t> Iterator for SubCapturesPos<'t> {
 /// name and the value.
 ///
 /// `'t` is the lifetime of the matched text.
-pub struct SubCapturesNamed<'t>{
-    caps: &'t Captures<'t>,
-    inner: Option<Iter<'t, String, usize>>,
+pub struct SubCapturesNamed<'c, 't: 'c> {
```

Comment:
Could we separate out the lifetime changes in public types into separate commits? The nice thing about the `Arc` hashmap is that we can land that today without any backwards incompatible changes I think.